presto-hive/src/test/java/io/prestosql/plugin/hive/benchmark/BenchmarkProjectionPushdownHive.java

```diff
@@ -0,0 +1,356 @@
+    private static final String WITH_PUSHDOWN = "with_pushdown";
+    private static final String WITHOUT_PUSHDOWN = "without_pushdown";
+
+    // Types
+    private static final String ROW_OF_STRINGS = "ROW(f0 VARCHAR, f1 VARCHAR, f2 VARCHAR)";
+    private static final String NESTED_STRUCT = "ROW(" +
+            "f0 VARCHAR, " +
+            "f1 VARCHAR, " +
+            "f2 VARCHAR, " +
+            "f3 ARRAY(ROW(f0 VARCHAR, f1 VARCHAR, f2 VARCHAR)))";
+
+    private final Random random = new Random();
+
+    private List<Type> columnTypesToWrite;
+    private List<String> columnNamesToWrite;
+    private List<HiveType> columnHiveTypesToWrite;
+
+    // Layout of the file columns: either in a single struct OR as "flattened" into top-level columns
+    @Param({STRUCT, TOP_LEVEL})
+    private String writeStrategy;
+
+    @Param({WITHOUT_PUSHDOWN, WITH_PUSHDOWN})
+    private String readStrategy;
+
+    // This should be a row typed column
+    @Param({ROW_OF_STRINGS, NESTED_STRUCT})
+    private String columnTypeString;
+
+    @Param("1")
+    private int readColumnCount;
+
+    @Param({"PRESTO_ORC", "PRESTO_PARQUET"})
+    private FileFormat fileFormat;
+
+    private TestData dataToWrite;
+    private File dataFile;
+
+    private final File targetDir = createTempDir("presto-benchmark");
+
+    @Setup
+    public void setup()
+            throws IOException
+    {
+        Metadata metadata = createTestMetadataManager();
+        Type columnType = metadata.fromSqlType(columnTypeString);
+        checkState(columnType instanceof RowType, "expected column to have RowType");
+
+        if (STRUCT.equals(writeStrategy)) {
+            columnTypesToWrite = ImmutableList.of(columnType);
+            columnHiveTypesToWrite = ImmutableList.of(toHiveType(columnType));
+            columnNamesToWrite = ImmutableList.of("column_0");
+        }
+        else if (TOP_LEVEL.equals(writeStrategy)) {
+            List<Type> fieldTypes = ((RowType) columnType).getTypeParameters();
+            columnTypesToWrite = ImmutableList.copyOf(fieldTypes);
+            columnHiveTypesToWrite = columnTypesToWrite.stream()
+                    .map(HiveType::toHiveType)
+                    .collect(toImmutableList());
+            columnNamesToWrite = IntStream.range(0, columnTypesToWrite.size())
+                    .mapToObj(Integer::toString)
+                    .map("column_"::concat)
+                    .collect(toImmutableList());
+        }
+        else {
+            throw new UnsupportedOperationException(format("Write strategy %s not supported", writeStrategy));
+        }
+
+        checkState(columnTypesToWrite.stream().allMatch(BenchmarkProjectionPushdownHive::isSupportedType), "Type not supported for benchmark");
+        dataToWrite = createTestData(columnTypesToWrite, columnNamesToWrite);
+
+        targetDir.mkdirs();
+        dataFile = new File(targetDir, UUID.randomUUID().toString());
+        writeData(dataFile);
+    }
+
+    @Benchmark
```

Comment:
please add a test that benchmark works, see `io.prestosql.operator.BenchmarkHashBuildAndJoinOperators#testBenchmarkBuildHash` for example